impl/src/main/java/io/jsonwebtoken/impl/compression/DefaultCompressionCodecResolver.java

```diff
@@ -54,19 +76,23 @@ public CompressionCodec resolveCompressionCodec(Header header) {
         if (!hasCompressionAlgorithm) {
             return null;
         }
-        if (io.jsonwebtoken.CompressionCodecs.DEFLATE.getAlgorithmName().equalsIgnoreCase(cmpAlg)) {
-            return io.jsonwebtoken.CompressionCodecs.DEFLATE;
-        }
-        if (io.jsonwebtoken.CompressionCodecs.GZIP.getAlgorithmName().equalsIgnoreCase(cmpAlg)) {
-            return io.jsonwebtoken.CompressionCodecs.GZIP;
-        }
-
-        throw new CompressionException("Unsupported compression algorithm '" + cmpAlg + "'");
+        return byName(cmpAlg);
     }
 
     private String getAlgorithmFromHeader(Header header) {
         Assert.notNull(header, "header cannot be null.");
 
         return header.getCompressionAlgorithm();
     }
+
+    private CompressionCodec byName(String name) {
+        Assert.hasText("'name' must not be empty");
```

Comment:
Looks like an invalid assertion - not passing in the `name` argument 😉

Reply:
🤦‍♂

Reply:
(also, good catch)